include/irr/asset/CMTLPipelineMetadata.h

```diff
@@ -0,0 +1,107 @@
+#ifndef __IRR_C_MTL_PIPELINE_METADATA_H_INCLUDED__
+#define __IRR_C_MTL_PIPELINE_METADATA_H_INCLUDED__
+
+#include "irr/asset/IPipelineMetadata.h"
+
+namespace irr {
+namespace asset
+{
+
+class CMTLPipelineMetadata final : public IPipelineMetadata
+{
+public:
+    struct SMtl
+    {
+        std::string name;
+
+#include "irr/irrpack.h"
+        struct
+        {
+            //Ka
+            core::vector3df_SIMD ambient = core::vector3df_SIMD(1.f);
+            //Kd
+            core::vector3df_SIMD diffuse = core::vector3df_SIMD(1.f);
+            //Ks
+            core::vector3df_SIMD specular = core::vector3df_SIMD(1.f);
+            //Ke
+            core::vector3df_SIMD emissive = core::vector3df_SIMD(1.f);
+            //Tf
+            core::vector3df_SIMD transmissionFilter = core::vector3df_SIMD(1.f);
+            //Ns, specular exponent in phong model
+            float shininess = 32.f;
+            //d
+            float opacity = 1.f;
+            //-bm
+            float bumpFactor = 1.f;
+
+            //PBR
+            //Ni, index of refraction
+            float IoR = 1.6f;
+            //Pr
+            float roughness = 0.f;
+            //Pm
+            float metallic = 0.f;
+            //Ps
+            float sheen = 0.f;
+            //Pc
+            float clearcoatThickness = 0.f;
+            //Pcr
+            float clearcoatRoughness = 0.f;
+            //aniso
+            float anisotropy = 0.f;
+            //anisor
+            float anisoRotation = 0.f;
+            //illum - bits [0;3]
+            //map presence: bits [4;16], order in accordance with E_MAP_TYPE
```

Comment:
you need 18 bits for the maps (just update the comment)

Reply:
nah, refl map takes 6 enum values but is just one map, that's why 13bits

Reply:
I see.